lib/GenomeFileUtil/core/FastaGFFToGenome.py

```diff
@@ -652,155 +548,145 @@ def _print_phytozome_gff(self, input_gff_file, feature_list):
-                        CDS_ftr["function"] = ""
-
-                    #Add to cdss array
-                    genome_cdss_list.append(CDS_ftr)
-
-                    #Add ids to gene_ftr arrays
-                    gene_ftr["cdss"].append(CDS_ftr["id"])
-
-                gene_ftr["protein_translation"] = longest_protein_sequence
-                gene_ftr["protein_translation_length"] = longest_protein_length
-                genome_features_list.append(gene_ftr)
+    def _transform_feature(self, contig, in_feature):
+        """Converts a feature from the gff ftr format into the appropriate
+        format for a genome object """
+        def _aliases(feat):
+            keys = ('locus_tag', 'old_locus_tag', 'protein_id',
+                    'transcript_id', 'gene', 'EC_number')
+            alias_list = []
+            for key in keys:
+                if key in feat['attributes']:
+                    alias_list.extend([(key, val) for val in feat['attributes'][key]])
+            return alias_list
+
+        def _propagate_cds_props_to_gene(cds, gene):
+            # Put longest protein_translation to gene
+            if "protein_translation" not in gene or (
+                len(gene["protein_translation"]) <
+                    len(cds["protein_translation"])):
+                gene["protein_translation"] = cds["protein_translation"]
+                gene["protein_translation_length"] = len(
+                    cds["protein_translation"])
+            # Merge cds list attributes with gene
+            for key in ('function', 'aliases', 'db_xref'):
+                if cds.get(key, []):
+                    gene[key] = cds.get(key, []) + gene.get(key, [])
+            # Merge cds["ontology_terms"] -> gene["ontology_terms"]
+            terms2 = cds.get("ontology_terms")
+            if terms2 is not None:
+                terms = gene.get("ontology_terms")
+                if terms is None:
+                    gene["ontology_terms"] = terms2
+                else:
+                    for source in terms2:
+                        if source in terms:
+                            terms[source].update(terms2[source])
+                        else:
+                            terms[source] = terms2[source]
+
+        # UTRs don't have any information not implied by the exon
+        excluded_features = ('five_prime_UTR', 'three_prime_UTR')
+        if in_feature['type'] in excluded_features:
+            self.skiped_features[in_feature['type']] += 1
+            return
+
+        # if the feature ID is duplicated we only need to update the location
+        if in_feature['ID'] in self.feature_dict:
+            self.feature_dict[in_feature['ID']]['location'].append(
+                self._location(in_feature))
+
+        self.feature_counts[in_feature['type']] += 1
+        if in_feature['start'] < 1 or in_feature['end'] > len(contig):
+            self.warn("Skipping feature with invalid location for specified "
```

Comment:
It seems like the contig or coordinates not being valid should be an error and not a warning.  If we want to add a "loose" mode in the future this could be a warning for that. But by default this should be an error.